### PR TITLE
test: Code Coverage: Fy-Select-Project Component

### PR DESCRIPTION
**Pull request by jayfyle**
## Edit 1: Coverage at 100%
fy_select_project_100

### Clickup: https://app.clickup.com/t/85zrxnafv

fy_select_project_94

### CODE REVIEW

File: src/app/shared/components/fy-select-project/fy-select-project.component.spec.ts [hunk 1]
```diff
@@ -1,26 +1,180 @@
...
+
+      component.writeValue(['value']);
+      expect(component.innerValue).toEqual(['value']);
+    });
+
+    it('should set display value to empty', () => {
+      component.innerValue = 'value';
+      fixture.detectChanges();
+
+      component.writeValue(undefined);
```
Comment (yksolanki9) [inline src/app/shared/components/fy-select-project/fy-select-project.component.spec.ts:131]: innerValue can be undefined. Calling writeValue with undefined should not happen.

File: src/app/shared/components/fy-select-project/fy-select-project.component.ts [hunk 2]
```diff
@@ -35,25 +35,23 @@ export class FySelectProjectComponent implements OnInit, ControlValueAccessor, O
 
   @Input() recentlyUsed: { label: string; value: ExtendedProject; selected?: boolean }[];
 
-  displayValue;
+  @Input() touchedInParent;
```
Comment (yksolanki9) [inline src/app/shared/components/fy-select-project/fy-select-project.component.ts:38]: Add types

File: src/app/shared/components/fy-select-project/fy-select-project.component.ts [hunk 3]
```diff
@@ -77,9 +75,7 @@ export class FySelectProjectComponent implements OnInit, ControlValueAccessor, O
     }
   }
 
-  ngOnInit() {
-    this.ngControl = this.injector.get(NgControl);
-  }
+  ngOnInit() {}
```
Comment (yksolanki9) [inline src/app/shared/components/fy-select-project/fy-select-project.component.ts:78]: Remove if unused

**Review by Julias0 — CHANGES_REQUESTED** [review]
Will review after Yash's review

File: src/app/shared/components/fy-select-project/fy-select-project.component.spec.ts [hunk 4]
```diff
@@ -1,26 +1,180 @@
...
+    projectModalSpy.onWillDismiss.and.returnValue(
+      Promise.resolve({
+        data: {
+          value: 'value1',
+        },
+      })
+    );
+    modalController.create.and.returnValue(Promise.resolve(projectModalSpy));
+
+    await component.openModal();
```
[the remark this answers is not in the document]
Comment (jayfyle) [reply src/app/shared/components/fy-select-project/fy-select-project.component.spec.ts:79]: I have a separate test case for checking if clicking on the DOM opens the modal. This is a test case specifically for checking if the method is working correctly

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
@jayfyle I don't see my comments being resolved

**Review by Julias0 — APPROVED** [review]

File: src/app/shared/components/fy-select-project/fy-select-project.component.spec.ts [hunk 5]
```diff
@@ -117,19 +119,19 @@ describe('FySelectProjectComponent', () => {
...
 
     it('should set display value to empty', () => {
-      component.innerValue = 'value';
+      component.innerValue = undefined;
       fixture.detectChanges();
 
-      component.writeValue(undefined);
-      expect(component.displayValue).toEqual('');
+      component.writeValue('value');
+      expect(component.displayValue).toEqual(undefined);
```
Comment (yksolanki9) [inline src/app/shared/components/fy-select-project/fy-select-project.component.spec.ts:134]: This should be empty string. How did this test even pass?

**Review by yksolanki9 — CHANGES_REQUESTED** [review]

File: src/app/shared/components/fy-select-project/fy-select-project.component.spec.ts (hunk 5 above)
Comment (jayfyle) [reply src/app/shared/components/fy-select-project/fy-select-project.component.spec.ts]: I have checked this test case and it does not return an empty string. The result is undefined.

**Review by yksolanki9 — APPROVED** [review]

**Pull request merged**